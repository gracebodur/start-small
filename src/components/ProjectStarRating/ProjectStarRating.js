import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export function ProjectStarRating({rating}) {
  const stars = [
    { filled: false },
    { filled: false },
    { filled: false },
    { filled: false },
    { filled: false },
  ]

  for (let i = 0; i < rating; i++) {
    stars[i].filled = true
  }

  return <span className='ProjectStarRating'>
    { stars.map((star, index) => <Star key={index} filled={star.filled} />) }
  </span>
}

function Star({filled}) {
  const library = filled ? 'fas' : 'far'
  return <FontAwesomeIcon className='yellow' icon={ [ library, 'star'  ]  } />
}
