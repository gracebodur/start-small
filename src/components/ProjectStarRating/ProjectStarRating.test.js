import React from "react";
import toJson from "enzyme-to-json";
import ProjectStarRating from "./ProjectStarRating";

describe(`ProjectStarRating component`, () => {
  const props = {
    id: "a",
    name: "test-class-name"
  };

  it("renders a ProjectStarRating by default", () => {
    const wrapper = <ProjectStarRating />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the ProjectStarRating given props", () => {
    const wrapper = <ProjectStarRating {...props} />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
