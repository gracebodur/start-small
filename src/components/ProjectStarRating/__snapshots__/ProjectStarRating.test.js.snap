// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectStarRating component renders a ProjectStarRating by default 1`] = `undefined`;

exports[`ProjectStarRating component renders the ProjectStarRating given props 1`] = `undefined`;
